src/main/java/com/minecolonies/coremod/entity/ai/citizen/deliveryman/EntityAIWorkDeliveryman.java

```diff
@@ -209,7 +209,7 @@ private boolean gatherFromBuilding(@NotNull final AbstractBuilding building)
         }
 
         if (workerRequiresItem(building, stack, alreadyKept)
-                || (building instanceof BuildingHome && stack.getItem() instanceof ItemFood))
+                || (building instanceof BuildingCook && !ItemStackUtils.isEmpty(stack) && stack.getItem() instanceof ItemFood))
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') A "NullPointerException" could be thrown; "stack" is nullable here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2259)